Composer/packages/client/src/components/ImportModal/ImportModal.tsx

```diff
@@ -0,0 +1,338 @@
+/** @jsx jsx */
+import { jsx, css } from '@emotion/core';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { navigate, RouteComponentProps } from '@reach/router';
+import { Dialog, DialogType, DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { DefaultButton, PrimaryButton } from 'office-ui-fabric-react/lib/Button';
+import formatMessage from 'format-message';
+import { FontWeights } from 'office-ui-fabric-react/lib/Styling';
+import { Spinner, SpinnerSize } from 'office-ui-fabric-react/lib/Spinner';
+
+import { ImportStatus } from './ImportStatus';
+import { useRecoilValue } from 'recoil';
+import { dispatcherState } from '../../recoilModel';
+import { createNotification } from '../../recoilModel/dispatchers/notification';
+import { ImportSuccessNotification } from './ImportSuccessNotification';
+
+type ImportedProjectInfo = {
+  alias?: string;
+  description?: string;
+  eTag: string;
+  name?: string;
+  source: string;
+  templateDir: string;
+  urlSuffix?: string;
+};
+
+type ImportPayload = {
+  name: string;
+  description?: string;
+};
+
+type ExistingProjectInfo = {
+  id: string;
+  location: string;
+  name: string;
+};
+
+type ImportModalState =
+  | 'copyingContent' // after selecting import to existing project
+  | 'connecting' // before login prompt shows
+  | 'downloadingContent' // after logging in
+  | 'failed' // error
+  | 'promptingToSave' // when an existing project is found under the same alias
+  | 'signingIn'; // when login prompt is showing
+
+const boldText = css`
+  font-weight: ${FontWeights.semibold};
+  color: #106ebe;
+  word-break: break-work;
+`;
+
+const CONNECTING_STATUS_DISPLAY_TIME = 2000;
+
+export const ImportModal: React.FC<RouteComponentProps> = (props) => {
+  const { location } = props;
+  const [importSource, setImportSource] = useState<string>('');
+  const [importPayload, setImportPayload] = useState<ImportPayload>({ name: '', description: '' });
+  const [importedProjectInfo, setImportedProjectInfo] = useState<ImportedProjectInfo | undefined>(undefined);
+  const [modalState, setModalState] = useState<ImportModalState>('connecting');
+  const [existingProject, setExistingProject] = useState<ExistingProjectInfo | undefined>(undefined);
+  const [error, setError] = useState<any>(undefined);
+  const [backupLocation, setBackupLocation] = useState<string>('');
+  const { addNotification } = useRecoilValue(dispatcherState);
+
+  const importAsNewProject = useCallback((info: ImportedProjectInfo) => {
+    // navigate to creation flow with template selected
```

Comment:
Have you considered passing this data as state to ```navigate``` function instead of the query params?
The reason I'm asking is that this url pattern is being only used to pass the data (e.g., url is not going to be used to share a deep linked content)

Reply:
I had no idea this was a thing 😮 I think that's a way better approach.